python/lsst/scarlet/lite/io.py

```diff
@@ -0,0 +1,598 @@
+            "factorized": [component.as_dict() for component in self.factorized_components],
+            "peak_id": self.peak_id,
+        }
+        return result
+
+    @classmethod
+    def from_dict(cls, data: dict, dtype: DTypeLike = np.float32) -> ScarletSourceData:
+        """Reconstruct `ScarletSourceData` from JSON compatible
+        dict.
+
+        Parameters
+        ----------
+        data:
+            Dictionary representation of the object
+        dtype:
+            Datatype of the resulting model.
+
+        Returns
+        -------
+        result:
+            The reconstructed object
+        """
+        data_shallow_copy = dict(data)
+        del data_shallow_copy["factorized"]
+        components = []
+        for component in data["components"]:
+            component = ScarletComponentData.from_dict(component, dtype=dtype)
+            components.append(component)
+        data_shallow_copy["components"] = components
+
+        factorized = []
+        for component in data["factorized"]:
+            component = ScarletFactorizedComponentData.from_dict(component, dtype=dtype)
+            factorized.append(component)
+        data_shallow_copy["factorized_components"] = factorized
+        data_shallow_copy["peak_id"] = int(data["peak_id"])
+        return cls(**data_shallow_copy)
+
+
+@dataclass
+class ScarletBlendData:
+    """Data for an entire blend.
+
+    Attributes
+    ----------
+    origin:
+        The lower bound of the blend's bounding box.
+    shape:
+        The shape of the blend's bounding box.
+    sources:
+        Data for the sources contained in the blend.
+    psf_center:
+        The location used for the center of the PSF for
+        the blend.
+    """
+
+    origin: tuple[int, int]
+    shape: tuple[int, int]
+    sources: dict[int, ScarletSourceData]
+    psf_center: tuple[float, float]
+    psf: np.ndarray
+
+    def as_dict(self) -> dict:
+        """Return the object encoded into a dict for JSON serialization
+
+        Returns
+        -------
+        result:
+            The object encoded as a JSON compatible dict
+        """
+        result = {
+            "origin": self.origin,
+            "shape": self.shape,
+            "psf_center": self.psf_center,
+            "psf_shape": self.psf.shape,
+            "psf": tuple(self.psf.flatten().astype(float)),
+            "sources": {bid: source.as_dict() for bid, source in self.sources.items()},
+        }
+        return result
+
+    @classmethod
+    def from_dict(cls, data: dict, dtype: DTypeLike = np.float32) -> ScarletBlendData:
+        """Reconstruct `ScarletBlendData` from JSON compatible
+        dict.
+
+        Parameters
+        ----------
+        data:
+            Dictionary representation of the object
+        dtype:
+            Datatype of the resulting model.
+
+        Returns
+        -------
+        result:
+            The reconstructed object
+        """
+        data_shallow_copy = dict(data)
```

Comment:
You're right that this could be done that way, perhaps it should be done that way. This is just the way the Nate originally coded up the model data classes and I never gave a second thought to changing it.